rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -112,13 +112,31 @@ class Player : public rclcpp::Node
   ROSBAG2_TRANSPORT_PUBLIC
   bool set_rate(double);
 
+  /// \brief Playing next message from queue when in pause.
+  /// \details This is blocking call and it will wait until next available message will be
+  /// published or rclcpp context shut down.
+  /// \note If internal player queue is starving and storage has not been completely loaded,
+  /// this method will wait until new element will be pushed to the queue.
+  /// It will be undefined behaviour if playback will be resumed from another thread during
```

Comment:
Not exactly. But possible to get rid from this constraint. I will move `skip_message_in_main_play_loop_mutex_` lock  at the very beginning of the `play_next()` to avoid race conditions.